flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -349,22 +380,42 @@ void testProperCleanupOnLeaderElectionCloseWhenHoldingTheLeadership() throws Exc
                             final UUID leaderSessionID = UUID.randomUUID();
                             grantLeadership(leaderSessionID);
 
-                            assertThat(testingContender.getLeaderSessionID()).isNotNull();
-                            assertThat(leaderElectionService.getLeaderSessionID(contenderID))
-                                    .isEqualTo(leaderSessionID);
-                            assertThat(storedLeaderInformation.get().forContenderID(contenderID))
-                                    .hasValue(LeaderInformation.known(leaderSessionID, TEST_URL));
+                            applyToBothContenderContexts(
+                                    ctx -> {
+                                        assertThat(ctx.contender.getLeaderSessionID())
+                                                .isEqualTo(leaderSessionID);
+                                        assertThat(
+                                                        leaderElectionService.getLeaderSessionID(
+                                                                ctx.contenderID))
+                                                .isEqualTo(leaderSessionID);
+
+                                        assertThat(
+                                                        leaderElectionService.getLeaderSessionID(
+                                                                ctx.contenderID))
+                                                .isEqualTo(leaderSessionID);
+
+                                        assertThat(
+                                                        storedLeaderInformation
+                                                                .get()
+                                                                .forContenderID(ctx.contenderID))
+                                                .hasValue(
+                                                        LeaderInformation.known(
+                                                                leaderSessionID, ctx.address));
```

Comment:
look at that magnificent formatting magic, going from 5 lines 18.

Reply:
yeah, I tried to reduce it already by providing the context members in the callback function instead of the context struct itself. That way, we would avoid having `ctx.` in the code. But it' doesn't help as much as I hoped.